autoblocks/_impl/tracerv2/run.py

```diff
@@ -0,0 +1,263 @@
+    # of propagating them to the caller. this is similar in behavior to Promise.allSettled
+    await asyncio.gather(*(sem_coro(c) for c in coroutines), return_exceptions=True)
+
+
+class AutoblocksTracer:
+    def __init__(
+        self,
+        ingestion_key: Optional[str] = None,
+        # Initialize the tracer with a trace_id. All events sent with this tracer will
+        # send this trace ID unless overwritten by:
+        # - calling set_trace_id
+        # - specifying a trace_id when calling send_event
+        trace_id: Optional[str] = None,
+        # Initialize the tracer with properties. All events sent with this tracer will
+        # send these properties unless overwritten by:
+        # - calling set_properties
+        # Additionally, these properties can be updated by:
+        # - calling update_properties
+        # - specifying properties when calling send_event
+        properties: Optional[Dict] = None,
+        # Timeout for sending events to Autoblocks
+        timeout: timedelta = timedelta(seconds=5),
+    ):
+        self._trace_id: Optional[str] = trace_id
+        self._properties: Dict = properties or {}
+
+        ingestion_key = ingestion_key or AutoblocksEnvVar.INGESTION_KEY.get()
+        if not ingestion_key:
+            raise ValueError(
+                f"You must provide an ingestion_key or set the {AutoblocksEnvVar.INGESTION_KEY} environment variable."
+            )
+
+        self._client = get_or_create_ingestion_client(ingestion_key)
+        self._timeout_seconds = timeout.total_seconds()
+        cli_server_address = AutoblocksEnvVar.CLI_SERVER_ADDRESS.get()
+        if not cli_server_address:
+            raise RuntimeError(
+                "Autoblocks tests must be run within the context of the testing CLI.\n"
+                "Make sure you are running your test command with:\n"
+                "$ npx autoblocks testing exec -- <your test command>"
+            )
+        self._loop = asyncio.new_event_loop()
+        background_thread = threading.Thread(
+            target=run_event_loop,
+            args=(loop,),
+            daemon=True,
+        )
+        background_thread.start()
+
+        self._started = True
+
+    async def _evaluate_event(
+        self,
+        event: Any,
+        evaluator: BaseEvaluator,
+    ):
+        evaluation: Optional[Evaluation] = None
+
+        if inspect.iscoroutinefunction(evaluator.evaluate):
+            try:
+                evaluation = await evaluator.evaluateOnline(event=event)
+            except Exception:
+                log.error("Do something here")
+        else:
+            try:
+                ctx = contextvars.copy_context()
+                evaluation = await loop.run_in_executor(
+                    None,
+                    ctx.run,
+                    evaluator.evaluate,
+                    event,
+                )
+            except Exception:
+                log.error("#TODO do something here")
```

Comment:
we probably just wanna log the exception and move on, we aren't going to want to re-raise